app/views/events/_schedule_tournament.html.erb

```diff
@@ -57,8 +57,12 @@
         <% end %>
       </td>
       <td>
-        <%= link_to t('events.schedule.insert_results'),
+        <%= link_to t('.show', default: t('helpers.links.show')),
                     match_path(match), :class => 'btn btn-default btn-xs' %>
+        <% if can? :delete, @match %>
```

Comment:
why ask permission for delete and redirect to edit path?

Reply:
No idea. Probably because it was too late... 
Should be `can? :edit, @match`